core/src/utilities/helpers/routing-helpers.js

```diff
@@ -23,8 +23,9 @@ export const getDefaultChildNode = async pathData => {
 
   if (lastElement.defaultChildNode && pathExists) {
     return lastElement.defaultChildNode;
-  } else if (children && children.length > 0) {
-    return children[0].pathSegment;
+  } else if (children && children.length) {
+    const validChild = children.find(child => child.pathSegment);
```

Comment:
If a node (except for the root one) has no pathSegment it is a clear misconfiguration. We should rather check this when loading the navigation configuration and either fail or remove this node from the tree. 
For the default child behaviour we should rather maybe check if the view Url is set and choose first node with viewUrl set?

Reply:
Ok, I'll do the part with checking `viewUrl` and `externalLink`. Let's leave the other part for a separate task.